roofit/roofitcore/inc/RooAbsAnaConvPdf.h

```diff
@@ -75,7 +75,12 @@ class RooAbsAnaConvPdf : public RooAbsPdf {
                                        const RooArgSet* auxProto=0, Bool_t verbose= kFALSE) const ;
   virtual Bool_t changeModel(const RooResolutionModel& newModel) ;
 
-  const RooRealVar* convVar() const ;  //  Convolution variable 
+  /// Retrieve the convolution variable.
+  const RooAbsRealLValue* convVar() const;
+  /// Retrieve the convolution variable.
+  RooAbsRealLValue* convVar() {
+    return const_cast<RooAbsRealLValue*>(static_cast<const RooAbsAnaConvPdf*>(this)->convVar());
```

Comment:
How's the`static_cast` not a `const_cast`? Am I forgetting something here?

Reply:
What you missed is that you need to call the const version of the function.
```
RooAbsAnaConvPdf.h:81:31: warning: all paths through this function will call itself [-Winfinite-recursion]
  RooAbsRealLValue* convVar() {
                              ^
1 warning generated.
```
The static cast avoids that. The const cast removes the constness. I agree, though, that it's easier to digest if I switch the implementations of the const and the non-const version, and only use a single const_cast.